common/cluster/archival_status.go

```diff
@@ -87,6 +82,16 @@ func (a *ArchivalConfig) EnableReadFromArchival() bool {
 	return a.enableReadFromArchival
 }
 
+func (a *ArchivalConfig) isDomainDefaultValid() bool {
+	if a.domainDefaultStatus == ArchivalPaused {
```

Comment:
Lets create separate types for clusterArchivalStatus and domainArchivalStatus. I think it will improve clarity as we later plumb this config throughout the code.